plugin-server/src/worker/ingestion/event-pipeline/2-emitToBufferStep.ts

```diff
@@ -110,7 +111,7 @@ export function shouldSendEventToBuffer(
 
     // shouldBufferAnonymousEvents indicated buffering should happen
     // for all anonymous events, irrespective of the person existing or not
-    const sendToBuffer = (shouldBufferAnonymousEvents && isAnonymousEvent) || !processEventImmediately
+    const sendToBuffer = shouldBufferAnonymousEvents || !processEventImmediately
```

Comment:
No actually this is not good we'll literally send everything to buffer, that's not what we want.

```suggestion
    const processEventImmediately = person || isIdentifyingEvent
    const maybeBuffer = !(isMobileLibrary || isAnonymousEvent)
    const sendToBuffer = !processEventImmediately && (maybeBuffer || shouldBufferAnonymousEvents)
```